packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt

```diff
@@ -0,0 +1,137 @@
+  private val bundledConstants: Bundle
+    get() {
+      val locale = Locale.getDefault()
+      val locales = locales
+      val localeNames = getLocaleNames(locales)
+      val isRTL = TextUtils.getLayoutDirectionFromLocale(locale) == View.LAYOUT_DIRECTION_RTL
+      val region = getRegionCode(locale)
+      val symbols = DecimalFormatSymbols(locale)
+      return Bundle().apply {
+        putString("currency", getCurrencyCode(locale))
+        putString("decimalSeparator", symbols.decimalSeparator.toString())
+        putString("digitGroupingSeparator", symbols.groupingSeparator.toString())
+        putStringArrayList("isoCurrencyCodes", iSOCurrencyCodes)
+        putBoolean("isMetric", !USES_IMPERIAL.contains(region))
+        putBoolean("isRTL", isRTL)
+        putString("locale", localeNames[0])
+        putStringArrayList("locales", localeNames)
+        putString("region", region)
+        putString("timezone", TimeZone.getDefault().id)
+      }
+    }
+  private val locales: ArrayList<Locale>?
+    get() {
+      val locales = ArrayList<Locale>()
+      val context = applicationContext ?: return null
+      val configuration = context.resources.configuration
+      if (VERSION.SDK_INT > VERSION_CODES.N) {
+        val localeList = configuration.locales
+        for (i in 0 until localeList.size()) {
+          locales.add(localeList[i])
+        }
+      } else {
+        locales.add(configuration.locale)
+      }
+      return locales
+    }
+
+  private fun getRegionCode(locale: Locale): String? {
+    val miuiRegion = getSystemProperty("ro.miui.region")
+    return if (!TextUtils.isEmpty(miuiRegion)) {
+      miuiRegion
+    } else getCountryCode(locale)
+  }
+
+  companion object {
+    private val USES_IMPERIAL = listOf("US", "LR", "MM")
+    private val iSOCurrencyCodes: ArrayList<String>
+      get() {
+        val locales = ArrayList<String>()
+        val availableCurrencies = Currency.getAvailableCurrencies()
+        for (handle in availableCurrencies) {
+          locales.add(handle.currencyCode)
+        }
+        return locales
+      }
+
+    private fun getLocaleNames(locales: ArrayList<Locale>?): ArrayList<String> {
+      val languages = ArrayList<String>()
+      for (locale in locales!!) {
+        // https://stackoverflow.com/a/46652446/4047926
+        languages.add(locale.toLanguageTag())
+      }
+      return languages
+    }
+
+    private fun getCountryCode(locale: Locale): String? {
+      return try {
+        val country = locale.country
+        if (TextUtils.isEmpty(country)) null else country
+      } catch (ignored: Exception) {
+        null
+      }
+    }
+
+    private fun getSystemProperty(key: String): String {
+      return try {
+        val systemProperties = Class.forName("android.os.SystemProperties")
+        val get = systemProperties.getMethod("get", String::class.java)
+        get.invoke(systemProperties, key) as String
+      } catch (ignored: Exception) {
+        ""
+      }
```

Comment:
Another possible way 😉 but up to you, might be less obvious than try-catch
```suggestion
      return runCatching {
        val systemProperties = Class.forName("android.os.SystemProperties")
        val get = systemProperties.getMethod("get", String::class.java)
        get.invoke(systemProperties, key) as String
      }.getOrNull() ?: ""
```